doc/user-guide/data-structures.rst

```diff
@@ -495,6 +495,201 @@ dimension and non-dimension variables:
+the constructor of the second:
+
+.. ipython:: python
+
+    ds2 = xr.Dataset({"bar": 0}, coords={"y": ("y", [0, 1, 2])})
+    # add a child by referring to the parent node
+    node2 = xr.DataTree(name="a", parent=dt, data=ds2)
+
+or by dynamically updating the attributes of one node to refer to another:
+
+.. ipython:: python
+
+    # add a second child by first creating a new node ...
+    ds3 = xr.Dataset({"zed": np.NaN})
+    node3 = xr.DataTree(name="b", data=ds3)
+    # ... then updating its .parent property
+    node3.parent = dt
+
+Our tree now has three nodes within it:
+
+.. ipython:: python
+
+    dt
+
+It is at tree construction time that consistency checks are enforced. For
+instance, if we try to create a `cycle` the constructor will raise an error:
+
+.. ipython:: python
+    :okexcept:
+
+    dt.parent = node3
+
+Alternatively you can also create a ``DataTree`` object from
+
+- An ``xarray.Dataset`` using ``Dataset.to_node()`` (not yet implemented),
+- A dictionary mapping directory-like paths to either ``DataTree`` nodes or
+  data, using :py:meth:`xarray.DataTree.from_dict()`,
+- A netCDF or Zarr file on disk with :py:func:`open_datatree()`. See
+  :ref:`reading and writing files <io>`.
+
+
+DataTree Contents
+~~~~~~~~~~~~~~~~~
+
+Like ``xarray.Dataset``, ``xarray.DataTree`` implements the python mapping interface,
+but with values given by either ``xarray.DataArray`` objects or other
+``DataTree`` objects.
+
+.. ipython:: python
+
+    dt["a"]
+    dt["foo"]
+
+Iterating over keys will iterate over both the names of variables and child nodes.
+
+We can also access all the data in a single node through a dataset-like view
+
+.. ipython:: python
+
+    dt["a"].ds
+
+This demonstrates the fact that the data in any one node is equivalent to the
+contents of a single ``xarray.Dataset`` object. The ``DataTree.ds`` property
+returns an immutable view, but we can instead extract the node's data contents
+as a new (and mutable) ``xarray.Dataset`` object via
+:py:meth:`xarray.DataTree.to_dataset()`:
+
+.. ipython:: python
+
+    dt["a"].to_dataset()
+
+Like with ``Dataset``, you can access the data and coordinate variables of a
+node separately via the ``data_vars`` and ``coords`` attributes:
+
+.. ipython:: python
+
+    dt["a"].data_vars
+    dt["a"].coords
+
+
+Dictionary-like methods
+~~~~~~~~~~~~~~~~~~~~~~~
+
+We can update a datatree in-place using Python's standard dictionary syntax,
+similar to how we can for Dataset objects. For example, to create this example
+datatree from scratch, we could have written:
+
+.. ipython:: python
+
+    dt = xr.DataTree(name="root")
+    dt["foo"] = "orange"
+    dt["a"] = xr.DataTree(data=xr.Dataset({"bar": 0}, coords={"y": ("y", [0, 1, 2])}))
+    dt["a/b/zed"] = np.NaN
+    dt
+
+To change the variables in a node of a ``DataTree``, you can use all the
+standard dictionary methods, including ``values``, ``items``, ``__delitem__``,
+``get`` and :py:meth:`xarray.DataTree.update`.
```

Comment:
I didn't want to resolve this without a comment...

I'm going to leave this as `:py:meth:`xarray.DataTree.update`, similar to the other related threads, so that the documentation displays `xarray.DataTree.update` instead of just `update()`.